lib/topological_inventory/ansible_tower/connection_manager.rb

```diff
@@ -0,0 +1,73 @@
+require "topological_inventory/ansible_tower/connection"
+require "receptor_controller-client"
+require "topological_inventory/ansible_tower/receptor/connection"
+
+module TopologicalInventory::AnsibleTower
+  # Connection manager provides connection either for public or on-premise towers
+  # depending on provided parameters
+  class ConnectionManager
+    include Logging
+
+    delegate :api_url, :to => :connection
+
+    attr_reader :connection
+
+    @@sync = Mutex.new
+    @@receptor_client = nil
```

Comment:
I think it would be better to make these class instance variables as opposed to class variables.
This will protect them from being inadvertently overridden from instances and subclasses.